server/util/tracer.js

```diff
@@ -0,0 +1,83 @@
+const { registerInstrumentations } = require('@opentelemetry/instrumentation');
+const opentelemetry = require('@opentelemetry/api');
+
+// Not functionally required but gives some insight what happens behind the scenes
+// const { diag, DiagConsoleLogger, DiagLogLevel } = opentelemetry;
+// diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.INFO);
+
+const { AlwaysOnSampler, SamplingDecision, SimpleSpanProcessor } = require('@opentelemetry/sdk-trace-base');
+const { GraphQLInstrumentation } = require('@opentelemetry/instrumentation-graphql');
+const { OTLPTraceExporter } = require('@opentelemetry/exporter-trace-otlp-proto');
+const { ConsoleSpanExporter, NodeTracerProvider } = require('@opentelemetry/sdk-trace-node');
+const { Resource } = require('@opentelemetry/resources');
+const { WinstonInstrumentation } = require('@opentelemetry/instrumentation-winston');
+const { SemanticResourceAttributes } = require('@opentelemetry/semantic-conventions');
+
+const Exporter = (process.env.EXPORTER || '').toLowerCase().startsWith('c') ? ConsoleSpanExporter : OTLPTraceExporter;
+const { ExpressInstrumentation } = require('@opentelemetry/instrumentation-express');
+const { HttpInstrumentation } = require('@opentelemetry/instrumentation-http');
+
+const otlpDisabled = process.env?.OTEL_SDK_DISABLED || false;
+const otlpEndpoint = process.env?.OTEL_EXPORTER_OTLP_ENDPOINT || null;
+
+// Filter Utility
+function filterSampler(filterFn, parent) {
+	return {
+		shouldSample(ctx, tid, spanName, spanKind, attr, links) {
+			if (!filterFn(spanName, spanKind, attr)) {
+				return { decision: SamplingDecision.NOT_RECORD };
+			}
+			return parent.shouldSample(ctx, tid, spanName, spanKind, attr, links);
+		},
+		toString() {
+			return `FilterSampler(${parent.toString()})`;
+		}
+	};
+}
+
+// Basic filter function, currently excludes internal express spans
+// attributes are helpful if we want to inspect and omit requests such as health checks or timesync
+// eslint-disable-next-line no-unused-vars
+function ignoreTheseSpans(spanName, spanKind, attributes) {
+	return spanKind !== opentelemetry.SpanKind.SERVER;
+}
+
+function setupTracing(serviceName) {
+	if (!otlpDisabled) {
+		const provider = new NodeTracerProvider({
+			resource: new Resource({
+				[SemanticResourceAttributes.SERVICE_NAME]: serviceName,
```

Comment:
We can set both of these with `OTLP_` environment variables. Would that be acceptable?

Reply:
What OTLP vars exists for the Ui pods at present?

Reply:
Ok added these in. OTEL_SERVICE_NAME exists but OTEL_SERVICE_NAMESPACE does not.